torch/csrc/profiler/util.cpp

```diff
@@ -402,6 +405,22 @@ std::unordered_map<std::string, std::string> saveNcclMeta(
   map.emplace(
       kGlobalRankStride, std::to_string(debugInfo->getGlobalRankStride()));
   map.emplace(kGroupSize, std::to_string(debugInfo->getWorldSize()));
+  map.emplace(kProcessGroupId, std::to_string(debugInfo->getProcessGroupId()));
+  auto& groupRanks = debugInfo->getGroupRanks();
+  if (!groupRanks.empty() && groupRanks.size() <= kTruncatLength) {
+    map.emplace(
+        kGroupRanks, fmt::format("\"[{}]\"", fmt::join(groupRanks, ", ")));
+  } else if (groupRanks.size() > kTruncatLength) {
+    map.emplace(
+        kGroupRanks,
+        fmt::format(
+            "\"[{}, ..., {}]\"",
+            fmt::join(
+                groupRanks.begin(),
+                groupRanks.begin() + kTruncatLength - 1,
```

Comment:
this probably does not need -1, since fmt::join iterative range is `[begin, end)`